go/systests/ephemeral_test.go

```diff
@@ -347,3 +346,52 @@ func TestEphemeralNewUserEKAndTeamEKAfterRevokes(t *testing.T) {
 	_, err = ephemeral.ForcePublishNewTeamEKForTesting(context.Background(), annG, teamID, *merkleRoot)
 	require.NoError(t, err)
 }
+
+func readdToTeamWithEKs(t *testing.T, leave bool) {
+	tt := newTeamTester(t)
+	defer tt.cleanup()
+
+	user1 := tt.addUser("one")
+	user2 := tt.addUser("wtr")
+
+	teamID, teamName := user1.createTeam2()
+	user1.addTeamMember(teamName.String(), user2.username, keybase1.TeamRole_WRITER)
+
+	ephemeral.ServiceInit(user1.tc.G)
+	ekLib := user1.tc.G.GetEKLib()
+	teamEK, err := ekLib.GetOrCreateLatestTeamEK(context.Background(), teamID)
+	require.NoError(t, err)
+
+	teamEKGeneration := teamEK.Metadata.Generation
+	var expectedGeneration keybase1.EkGeneration
+	if leave {
+		user2.leave(teamName.String())
+		expectedGeneration = teamEKGeneration
+	} else {
+		user1.removeTeamMember(teamName.String(), user2.username)
+		expectedGeneration = teamEKGeneration + 1
+	}
+
+	// After leaving user2 won't have access to the current teamEK
```

Comment:
my worry here is that this code can race user1's rotate key  CLKR handler after leave